docs/extension-system/_index.md

```diff
@@ -0,0 +1,148 @@
+
+#### Extension Registry
+
+The globally available extension registry provided by the ownCloud Web runtime can be used to both register and query extensions. All extensions
+which are being made available via an `app` get registered in the extension registry automatically. In your custom application code you can
+then query any of the available extensions by providing a `type` string and optionally a list of `scopes`. Throughout the ownCloud Web platform
+and most prominently also in the `files` app we have defined some extension points which automatically use certain extensions, see the 
+`Extension Points` section below.
+
+#### Extension Types
+
+Building an extension is limited to the extension types that are defined by the ownCloud Web extension system. See the full list of available extension types below.
+
+1. `ActionExtension` (type `action`) - An extension that can register `Action` items which then get shown in various places (e.g. context menus, batch actions), depending on their 
+respective scope. Most commonly used for file and folder actions (e.g. copy, rename, delete, etc.). For details, please refer to the [action docs]({{< ref "extension-types/actions.md" >}})
+2. `SearchExtension` (type `search`) - An extension that can register additional search providers. For details, please refer to the [search docs]({{< ref "extension-types/search.md" >}})
+3. `SidebarNavExtension` (type `sidebarNav`) - An extension that can register additional navigation items to the left sidebar. These can be scoped to specific apps, and programmatically enabled/disabled.
+For details, please refer to the [sidebar nav docs]({{< ref "extension-types/left-sidebar-menu-item.md" >}})
+4. `SidebarPanelExtension`, (type `sidebarPanel`) - An extension that can register panels to the right sidebar. For details, please refer to the [sidebar panel docs]({{< ref "extension-types/right-sidebar-panels.md" >}})
+5. `FolderViewExtension` (type `folderView`) - An extension that can register additional ways of displaying the content of a folder (resources, so spaces, folders or files) to the user.
+For details, please refer to the [folder view docs]({{< ref "extension-types/folder-view.md" >}})
+
+You're free to introduce your own extension types within your application code and use the extension registry to query the available ones. However, if you have the impression
+that an important extension type is missing, please reach out to us by opening a [GitHub issue](https://github.com/owncloud/web/issues/new/choose).
+
+#### Extension Base Configuration
+
+Any extension is required to define at least an `id` and a `type`.
+
+The `id` is supposed to be unique throughout the ownCloud Web ecosystem. In order to keep `id`s readable for humans we didn't want to enforce uniqueness through e.g. uuids. 
+Instead, we chose to use dot-formatted namespaces like e.g. `com.github.owncloud.web.files.search`. We'd like to encourage you to follow the same format for your own extensions.
```

Comment:
Maybe worth dropping a line why we require it (basically to be able to disable specific extensions via global config)